tracer/src/Datadog.Trace/Propagators/ParseUtility.cs

```diff
@@ -153,6 +153,69 @@ static bool TryParse(IEnumerable<string?> headerValues, ref bool hasValue, out i
             }
         }
 
+        public static ulong? ParseUInt64<TCarrier>(TCarrier headers, string headerName)
+            where TCarrier : IHeadersCollection
```

Comment:
I think we're moving _away_ from using `IHeadersCollection` in favor of `TCarrierGetter`/`ICarrierGetter<TCarrier>`. We used to have all these `ParseXYZ()` methods before and removed them 😅 

https://github.com/DataDog/dd-trace-dotnet/blob/d939f81bdfbf85272b349ca1899590d2b76b2707/tracer/src/Datadog.Trace/SpanContextPropagator.cs#L261-L262
(I picked a random commit from 2021, there's nothing significant about this in in particular)

Reply:
Anything that is still using `IHeadersCollection` is legacy code.

Reply:
😅 Okay good thing that wasn't used as I totally went with the `IHeadersCollection` at first but opted for the `TCarrier` as it worked better